implementations/rust/ockam/ockam_app_lib/src/background_node.rs

```
@@ -85,37 +86,45 @@ impl BackgroundNodeClient for Cli {
 #[async_trait]
 impl Nodes for Cli {
     async fn create(&mut self, node_name: &str) -> Result<()> {
-        duct::cmd!(
-            &self.bin,
-            "--no-input",
-            "node",
-            "create",
-            node_name,
-            "--trust-context",
-            node_name
-        )
-        .before_spawn(log_command)
-        .stderr_null()
-        .stdout_capture()
-        .run()?;
+        {
+            let bin = self.bin.clone();
+            let node_name = node_name.to_string();
+            spawn_blocking(move || {
+                duct::cmd!(
+                    &bin,
+                    "--no-input",
+                    "node",
+                    "create",
+                    &node_name,
+                    "--trust-context",
+                    &node_name
+                )
+                .before_spawn(log_command)
+                .stderr_null()
+                .stdout_capture()
+                .run()
+            })
+            .await
+            .unwrap()?;
```

Perhaps, instead of returning the result from duct just to unwrap it, and since we can't recover from that failure, we could do the following:

```rust
spawn_blocking(move || {
    let _ = duct::cmd!(
        &bin,
        "--no-input",
        "node",
        "create",
        &node_name_moved,
        "--trust-context",
        &node_name_moved
    )
    .before_spawn(log_command)
    .stderr_null()
    .stdout_capture()
    .run()
    .map(|_| debug!(node = %node_name, "Node created"));
})
.await?;
```

---

done! 👍